stan/math/prim/fun/multiply.hpp

```diff
@@ -126,7 +150,7 @@ inline auto multiply(const Mat1& m1, const Mat2& m2)
  * @throw <code>std::invalid_argument</code> if rv and v are not the same size
  */
 template <typename RowVec, typename ColVec,
-          require_all_not_st_var<RowVec, ColVec>* = nullptr,
+          require_not_t<is_var<return_type_t<RowVec, ColVec>>>* = nullptr,
```

Comment:
I like `require_all_not_st_var` honestly

Reply:
I switched this to `require_not_var_t<return_type_t<RowVec, ColVec>>`. The issue here is that a vector of vars can be allowed here if the other type is complex.